### PR TITLE
Change Transcode API Task output format

### PR DESCRIPTION
**Pull request by leszko**
Related discussion in https://github.com/livepeer/catalyst/issues/397

### CODE REVIEW

File: task/upload.go [hunk 1]
```diff
@@ -148,20 +149,66 @@ func TaskTranscodeFile(tctx *TaskContext) (*TaskHandlerOutput, error) {
...
+
+	p := u.Path
+	if strings.HasPrefix(u.Scheme, "s3+http") {
+		// first part of the Object Store path is a bucket name, skip it
+		ps := strings.Split(strings.TrimLeft(p, "/"), "/")
+		p = "/" + strings.Join(ps[1:], "/")
+	}
+
+	var baseUrl string
+	if strings.HasPrefix(u.Scheme, "ipfs") {
```
Comment (victorges) [inline task/upload.go:203]: I think this should be an exact check instead wdyt? Is there any other protocol that starts with `ipfs`?
```suggestion
	if u.Scheme == "ipfs" {
```

File: task/upload_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+import (
+	"github.com/livepeer/catalyst-api/video"
+	"github.com/livepeer/livepeer-data/pkg/data"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestToTranscodeFileTaskOutput(t *testing.T) {
+	tests := []struct {
```
Comment (victorges) [inline task/upload_test.go:11]: Can you add a case without MP4 outputs as well?

**Review by victorges — APPROVED** [review]
LGTM, only some suggestions

File: task/runner.go [hunk 3]
```diff
@@ -382,11 +383,11 @@ func (r *runner) HandleCatalysis(ctx context.Context, taskId, nextStep, attemptI
 		}
 	}
 
-	if callback.Status == clients.CatalystStatusError {
+	if callback.Status == catalystClients.TranscodeStatusError {
```
[the remark this answers is not in the document]
Comment (leszko) [reply task/runner.go:386]: Good idea, removed.

File: task/upload.go [hunk 4]
```diff
@@ -148,20 +149,66 @@ func TaskTranscodeFile(tctx *TaskContext) (*TaskHandlerOutput, error) {
...
 		},
 		profiles:                 params.Profiles,
 		catalystPipelineStrategy: pipeline.Strategy(params.CatalystPipelineStrategy),
 	})
 }
 
+func toTranscodeFileTaskOutput(outputs []video.OutputVideo) (data.TranscodeFileTaskOutput, error) {
+	var res data.TranscodeFileTaskOutput
+
+	if outputs == nil || len(outputs) < 1 {
```
[the remark this answers is not in the document]
Comment (leszko) [reply task/upload.go:166]: Thanks!

File: task/upload.go [hunk 5]
```diff
@@ -148,20 +149,66 @@ func TaskTranscodeFile(tctx *TaskContext) (*TaskHandlerOutput, error) {
...
+}
+
+func parseUrlToBaseAndPath(URL string) (string, string, error) {
+	u, err := url.Parse(URL)
+	if err != nil {
+		return "", "", err
+	}
+
+	p := u.Path
+	if strings.HasPrefix(u.Scheme, "s3+http") {
```
[the remark this answers is not in the document]
Comment (leszko) [reply task/upload.go:196]: Yeah, agree with you, we need some helpers functions. Added a GH Issue here: https://github.com/livepeer/go-tools/issues/23

Saying that, I think string check is actually better then type-cast check, so for now, let's leave it as it is. We'll refactor while working on the related issue.

File: task/upload_test.go (hunk 2 above)
Comment (leszko) [reply task/upload_test.go:11]: Good idea, added.

File: task/upload.go [hunk 6]
```diff
@@ -148,20 +149,66 @@ func TaskTranscodeFile(tctx *TaskContext) (*TaskHandlerOutput, error) {
...
+	if err != nil {
+		return res, err
+	}
+
+	for _, m := range o.MP4Outputs {
+		_, p, err := parseUrlToBaseAndPath(m.Location)
+		if err != nil {
+			return res, err
+		}
+		res.Mp4.Renditions = append(res.Mp4.Renditions, data.TranscodeFileTaskOutputPath{Path: p})
```
[the remark this answers is not in the document]
Comment (leszko) [reply task/upload.go]: I think it's not needed after we got rid of `Renditions`

**Pull request merged**